example/src/App.tsx

```diff
@@ -427,7 +408,28 @@ export default function App() {
       };
       cleanupAsync();
     };
-  }, [roomChanged, localAudioTrack, localVideoTrack]);
+  }, [roomChanged]);
+
+  const localParticipant = room?.localParticipant;
+  useEffect(() => {
+    if (localParticipant && localVideoTrack) {
+      localParticipant.publishLocalVideoTrack(localVideoTrack);
+      return () => {
+        localParticipant.unpublishLocalVideoTrack(localVideoTrack);
```

Comment:
this errors every time, for reasons I've described in slack.

full sequence of events for this hook:
- on initial render, we don't have a `localParticipant` and `localVideoTrack`, pass
- the `LocalVideoTrack.create` hook completes, triggers a render here
- we publish that local video track
- `deviceId` changes
- the `LocalVideoTrack.create` hook cleanup sends the currently published `LocalVideoTrack` a `destroy` message, which unpublishes it from all local participants in kotlinland
- then we get here, and this errors out when it can't find the already destroyed track

should `unpublishLocalVideoTrack` just no-op on 404?
should we have the android app hang on to the names of previously destroyed tracks, so we can have it no-op in this case in particular? what about if a user reuses a track name?
should we enforce that in order to destroy a track, the user must first unpublish it in js land?

my preference is for this operation to no-op if the local video track doesn't exist in kotlinland or isn't currently published by the local participant, but I'm not sure that's great

Reply:
> my preference is for this operation to no-op if the local video track doesn't exist in kotlinland or isn't currently published by the local participant, but I'm not sure that's great

i don't really see that much downside to this approach. 

it feels like either destroy should also unpublish or unpublish should also destroy (or both), so that those operations can be more "atomic" than they are now.

Reply:
talked in slack about this, but went with this as a solution: https://github.com/kfinn/react-native-twilio-video/pull/3/files#diff-3ca53efd7cd2ad8d59ebec10aebed074f72d644863cdc01c02b06a7981673be1R128